IndicoV2/Storage/MultipartFormUpload.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace IndicoV2.Storage
+{
+    internal class MultipartFormUpload : IRestRequest<JArray>
+    {
+        private readonly IndicoClient _client;
+        public List<FileParameter> FileParameters { get; set; }
+
+        public MultipartFormUpload(IndicoClient client) => _client = client;
+
+        public async Task<JArray> Call(CancellationToken cancellationToken = default)
+        {
+            string uploadUrl = _client.BaseUri + "/storage/files/store";
+            var formData = await MultipartFormDataContent(FileParameters);
+            var client = _client.HttpClient;
+            var responseMessage = await client.PostAsync(uploadUrl, formData, cancellationToken);
+            string body = await responseMessage.Content.ReadAsStringAsync();
+            var uploadResult = JArray.Parse(body);
+            return uploadResult;
+        }
+
+        private async Task<HttpContent> MultipartFormDataContent(List<FileParameter> parameters)
```

Comment:
Hmm.. might be worth making a ticket and seeing if there's a better way to do this in the future, I feel like handling multipart form data boundaries maybe isn't something we should need to do by hand (not necessary to address for this task tho)